pkg/antctl/command_definition_test.go

```diff
@@ -439,3 +442,711 @@ func TestCommandDefinitionGenerateExample(t *testing.T) {
+							},
+							{
+								name:      "namespace",
+								usage:     "Get log-level from specific Namespace.",
+								shorthand: "na",
+							},
+							{
+								name:      "output",
+								usage:     "used to display output",
+								shorthand: "out",
+							},
+						},
+					},
+				},
+				controllerEndpoint: &endpoint{},
+				agentEndpoint:      &endpoint{},
+			},
+			mode: "flowaggregator",
+			expectedErrors: []string{
+				"log-level: command does not define output struct",
+				"log-level: command for agent must define one endpoint",
+				"log-level: command for controller must define one endpoint",
+				"log-level: flag name cannot be empty",
+				"log-level: length of a flag shorthand cannot be larger than 1: na",
+				"log-level: flag redefined: output",
+				"log-level: length of a flag shorthand cannot be larger than 1: out",
+			},
+		},
+		{
+			name: "Command for flowaggregator defines resource endpoint",
+			cd: &commandDefinition{
+				use: "podmulticasts",
+				flowAggregatorEndpoint: &endpoint{
+					resourceEndpoint: &resourceEndpoint{
+						namespaced: true,
+					},
+				},
+				controllerEndpoint: &endpoint{},
+				agentEndpoint:      &endpoint{},
+			},
+			mode: "flowaggregator",
+			expectedErrors: []string{
+				"podmulticasts: command does not define output struct",
+				"podmulticasts: command for agent must define one endpoint",
+				"podmulticasts: command for controller must define one endpoint",
+			},
+		},
+	}
+	for _, tt := range tc {
+		t.Run(tt.name, func(t *testing.T) {
+			runtime.Mode = tt.mode
+			errs := tt.cd.validate()
+			strErrors := make([]string, len(errs))
+			for i, err := range errs {
+				strErrors[i] = err.Error()
+			}
+			assert.Equal(t, tt.expectedErrors, strErrors)
+		})
+	}
+}
+
+func TestGetRequestErrorFallback(t *testing.T) {
+	tc := []struct {
+		name     string
+		cd       *commandDefinition
+		expected int
+		mode     string
+	}{
+		{
+			name: "Antctl running against agent mode",
+			mode: "agent",
+			cd: &commandDefinition{
+				agentEndpoint: &endpoint{
+					requestErrorFallback: fallbackversion.RequestErrorFallback,
+				},
+			},
+			expected: 2,
+		},
+		{
+			name: "Antctl running against controller mode",
+			mode: "controller",
+			cd: &commandDefinition{
+				controllerEndpoint: &endpoint{
+					requestErrorFallback: fallbackversion.RequestErrorFallback,
+				},
+			},
+			expected: 2,
+		},
+		{
+			name: "Antctl running against flowaggregator mode",
+			mode: "flowaggregator",
+			cd: &commandDefinition{
+				flowAggregatorEndpoint: &endpoint{
+					requestErrorFallback: fallbackversion.RequestErrorFallback,
+				},
+			},
+			expected: 2,
+		},
+	}
+	for _, tt := range tc {
+		t.Run(tt.name, func(t *testing.T) {
+			runtime.Mode = tt.mode
+			fallback := tt.cd.getRequestErrorFallback()
+			reader, err := fallback()
+			assert.Nil(t, err)
+			out, err := reader.Read([]byte("test"))
+			assert.Nil(t, err)
+			assert.Equal(t, tt.expected, out)
```

Comment:
I don't get this part, what does `2` means here? 
And all your expected value are `2`, you may just compare it directly via `assert.Equal(t, 2, out)` without a field.